src/components/id/AddToCart/AddToCard.tsx

```diff
@@ -0,0 +1,56 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { Button } from "@mui/material";
+
+import { ProductType } from "@/types/product";
+
+const AddToCart = ({ data }: { data: ProductType }) => {
+  const [isInCart, setIsInCart] = useState(false);
+
+  useEffect(() => {
+    if (typeof window !== "undefined" && data && data.id) {
+      const cart = JSON.parse(localStorage.getItem("cart") || "[]");
+      const isProductInCart = cart.some(
+        (item: ProductType) => item.id === data.id
+      );
+      if (isProductInCart) {
+        setIsInCart(true);
+      }
+    }
+  }, [data]);
+
+  const handleAdd = () => {
+    if (typeof window !== "undefined" && data && data.id) {
+      let cart = JSON.parse(localStorage.getItem("cart") || "[]");
+
+      const isProductInCart = cart.some(
+        (item: ProductType) => item.id === data.id
+      );
+
+      if (isProductInCart) {
+        cart = cart.filter((item: ProductType) => item.id !== data.id);
+        setIsInCart(false);
```

Comment:
Here as well no point to check for window, again u repeat working with local storage and no custom hook :(